hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -545,33 +552,37 @@ class HoodieSparkSqlWriterInternal {
                          latestTableSchemaOpt: Option[Schema],
                          internalSchemaOpt: Option[InternalSchema],
                          opts: Map[String, String]): Schema = {
+    val addNullForDeletedColumns = opts.getOrDefault(DataSourceWriteOptions.ADD_NULL_FOR_DELETED_COLUMNS.key(),
```

Comment:
we definitely need good docs for this method. lets enhance docs in L545. even having illustrative examples is totally fine. We have been soft/low key on schema evolution in general. lets button up and ensure we get it right this time.

Reply:
the schema handling is grown up to be sizable now. Lets move it to a separate static class instead of adding everything to HoodieSparkSqlWrtier.

Reply:
I am moving deduceWriterSchema() to HoodieSchemaUtils